factory/cert_utils.go

```diff
@@ -92,7 +94,12 @@ func NewSignedCert(signer crypto.Signer, caCert *x509.Certificate, caKey crypto.
 		return nil, err
 	}
 
-	return x509.ParseCertificate(cert)
+	parsedCert, err := x509.ParseCertificate(cert)
+	if err == nil {
+		logrus.Infof("certificate %v signed by %v: notBefore=%v notAfter=%v",
```

Comment:
If any of the values being printed are supported, ie %s, %d, etc, the format should be updated accordingly.

Reply:
these are pkix.Name and time.Time. Is it preferred to use %s for non-string types or let %v handle the formatting?

Reply:
It's better to be explicit whenever possible.